src/server/scripts/EasternKingdoms/zone_ghostlands.cpp

```diff
@@ -0,0 +1,198 @@
+#include "AreaTriggerAI.h"
+#include "Containers.h"
+#include "Conversation.h"
+#include "ConversationAI.h"
+#include "CreatureAIImpl.h"
+#include "MotionMaster.h"
+#include "ObjectAccessor.h"
+#include "Player.h"
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "TemporarySummon.h"
+
+enum ToGhostlandsData
+{
+    NPC_SUMMONED_HIGH_ELF_RANGER        = 130919,
+
+    QUEST_TO_GHOSTLANDS                 = 49787,
+
+    CONVO_SANCTUM_OF_THE_MOON           = 6467,
+    CONVO_SANCTUM_OF_THE_MOON_FOLLOWUP  = 6634,
+    CONVO_ANDILIEN_ESTATE               = 6469,
+    CONVO_ANDILIEN_ESTATE_FOLLOWUP      = 6479,
+    CONVO_DAWNSPAIR_SPIRE               = 6470,
+    CONVO_DAWNSPAIR_SPIRE_UP            = 6637,
+    CONVO_DAWNSPAIR_SPIRE_FOLLOWUP      = 6471,
+
+    CONVO_ACTOR_HIGH_ELF_RANGER         = 61831
+};
+
+// 6468 - Conversation
+class conversation_high_elf_ranger_to_ghostlands : public ConversationAI
+{
+public:
+    using ConversationAI::ConversationAI;
+
+    void OnCreate(Unit* creator) override
+    {
+        Creature* highElfRanger = creator->FindNearestCreatureWithOptions(20.0f, { .CreatureId = NPC_SUMMONED_HIGH_ELF_RANGER, .IgnorePhases = true, .OwnerGuid = creator->GetGUID() });
+        if (!highElfRanger)
+            return;
+
+        conversation->AddActor(CONVO_ACTOR_HIGH_ELF_RANGER, 0, highElfRanger->GetGUID());
+        conversation->Start();
+    }
+};
+
+// 6478 - Conversation
+// 6635 - Conversation
+template<uint32 ConversationId>
+class conversation_ghostlands_start_convo_on_end : public ConversationAI
+{
+public:
+    using ConversationAI::ConversationAI;
+
+    void OnStart() override
+    {
+        conversation->m_Events.AddEvent([conversation = conversation]()
+        {
+            Player* player = ObjectAccessor::GetPlayer(*conversation, conversation->GetPrivateObjectOwner());
+            if (!player)
+                return;
+
+            Conversation::CreateConversation(ConversationId, player, *player, player->GetGUID(), nullptr, true);
+        }, conversation->GetLastLineEndTime(conversation->GetPrivateObjectOwnerLocale()));
+    }
+};
+
+constexpr Position ToGhostlandsAlleriaMoveToBook = { 7835.05f, -7913.11f, 308.2204f };
+
+// 6636 - Conversation
+class conversation_dawnspair_spire_to_ghostlands : public ConversationAI
+{
+public:
+    using ConversationAI::ConversationAI;
+
+    enum DawnspairSpireConvoData
+    {
+        POINT_ALLERIA_MOVE_TO_BOOK = 1
+    };
+
+    void OnCreate(Unit* creator) override
+    {
+        Creature* highElfRanger = creator->FindNearestCreatureWithOptions(20.0f, { .CreatureId = NPC_SUMMONED_HIGH_ELF_RANGER, .IgnorePhases = true, .OwnerGuid = creator->GetGUID() });
+        if (!highElfRanger)
+            return;
+
+        TempSummon* alleriaClone = highElfRanger->SummonPersonalClone(highElfRanger->GetPosition(), TEMPSUMMON_MANUAL_DESPAWN, 0s, 0, 0, creator->ToPlayer());
+        if (!alleriaClone)
+            return;
+
+        alleriaClone->SetWalk(true);
```

Comment:
Is there a SMSG_MOVE_SPLINE_SET_WALK_MODE (or MOVEMENTFLAG_WALKING in CreateObject) or is this here to only get proper movement speed for `MovePoint`? If it's for the movement only then you can pass a custom speed or tell it to use walk speed by default to `MovePoint`